cargo-shuttle/Cargo.toml

```diff
@@ -15,7 +15,7 @@ cargo = "0.65.0"
 cargo-edit = { version = "0.11.6", features = ["cli"] }
 cargo_metadata = "0.15.2"
 chrono = { workspace = true }
-clap = { version = "3.2.17", features = ["derive", "env"] }
+clap = { version = "4.0.29", features = ["derive", "env"] }
 clap_complete = "3.2.5"
```

Comment:
For the mismatched types error from the `generate` function, upgrading `clap_complete` to the latest should be enough.